spec/test_tree_visibility.rb

```diff
@@ -0,0 +1,15 @@
+require_relative '../solution/tree_visibility'
+
+RSpec.describe 'example test' do
+  it 'example1' do
+    t1 = TreeVisibility.new(5,
+                            TreeVisibility.new(3, TreeVisibility.new(20, nil, nil), TreeVisibility.new(21, nil, nil)), TreeVisibility.new(10, TreeVisibility.new(1, nil, nil), nil))
```

Comment:
Better readable version.

```suggestion
    t1 = TreeVisibility.new(
      5,
      TreeVisibility.new(
        3,
        TreeVisibility.new(20, nil, nil),
        TreeVisibility.new(21, nil, nil)
      ),
      TreeVisibility.new(10, TreeVisibility.new(1, nil, nil), nil)
    )
```